### ISSUE TITLE
danger process isn't exiting the process with the same error code as the subprocess

### ISSUE THREAD

**Issue by orta** [danger/danger-js#398]
I saw it "fail" a bunch, but actually pass. It even said "process exited with code '1'" or whatever.

**Comment by orta** [comment]
If danger can successfully set a commit status to be red, it won't fail the build. ( Whereas the `--fail-on-error` option would make it fail regardless ) 

Basically this is a guard against people having danger run first on their CI, and because you failed a CHANGELOG check or something it won't pass through to unit tests etc

If this isn't what's happening, then I definitely think it's abug

**Comment by JamieMason** [comment]
Understood re: not failing the build, I think we'd just misunderstood the difference between the CI build failing and the PR failing via danger. 

Regarding `--fail-on-error` / `--fail-on-errors` though: `danger ci` throws if you try to use that, saying it is not recognised?

We're using `"danger": "^3.3.2"` in package.json.

**Comment by orta** [comment]
Yeah, looks like that flag hasn't been moved over form the ruby version, I don't use it - so I've not ported it

```
~/dev/projects/artsy/js/sites/metaphysics validate_env
❯ danger ci --help

  Usage: danger-ci [options]

  Runs a Dangerfile in JavaScript or TypeScript.

  Options:

    -v, --verbose                            Verbose output of files
    -c, --external-ci-provider [modulePath]  Specify custom CI provider
    -t, --text-only                          Provide an STDOUT only interface, Danger will not post to your PR
    -d, --dangerfile [filePath]              Specify a custom dangerfile path
    -i, --id [danger_id]                     Specify a unique Danger ID for the Danger run
    -b, --base [branch_name]                 Base branch
    -h, --help                               output usage information
```

**Comment by orta** [comment]
I think the messaging could be improved about the process exit code IMO too

**Comment by weitzman** [comment]
This was recently added. See --failOnErrors
https://github.com/danger/danger-js/blob/master/source/commands/utils/sharedDangerfileArgs.ts#L27

**Comment by adityaridha** [comment]
Hi, need help on similar issue...
I'm running danger using NPM with `--failOnErrors` flag, but the job still a success
Any idea what's wrong with the implementation ?

````
{
  "dependencies": {
    "danger": "^10.7.1",
    "get-urls": "^10.0.1",
    "node-fetch": "^3.1.0",
    "pg": "^8.6.0",
    "pg-hstore": "^2.3.4",
    "sequelize": "^6.6.5"
  },
  "type": "module",
  "scripts": {
    "danger": "danger"
  }
}

```

![image](https://user-images.githubusercontent.com/16025454/148347533-e0c6e795-958d-466e-955c-724750e1afef.png)

**Comment by binary64** [comment]
I too am looking for danger to exit with a non-0 level when there is 1 or more fail()'s emitted. This is because I'm using --no-publish-check because I wanted a single unified build on my PRs.

**Comment by dpobel** [comment]
@binary64 it looks like `danger ci` has to be called with `--failOnErrors` so that it actually exit with a non 0 process exit status.

The _Failing the build_ message is really confusing, without this option there's no way for danger to actually fail a build…

**Comment by fbartho** [comment]
@kmannislands -- this ticket isn't about that case. If you'd like to propose the breaking change of defaulting `--failOnErrors` to be the standard behavior, feel free to file a separate ticket.

That said, this would be a breaking change, and it just hasn't been a priority since users can get the behavior they want today.
